src/http_handlers/metrics.hpp

```diff
@@ -77,45 +152,51 @@ class MetricsRequestHandler final {
   ~MetricsRequestHandler() = default;
 
   template <class Body, class Allocator>
-  void HandleRequest(http::request<Body, http::basic_fields<Allocator>> &&req,
-                     std::function<void(http::response<http::string_body>)> &&send) {
+  // NOLINTNEXTLINE(misc-unused-parameters)
+  void HandleRequest(boost::beast::http::request<Body, boost::beast::http::basic_fields<Allocator>> &&req,
+                     std::function<void(boost::beast::http::response<boost::beast::http::string_body>)> &&send) {
     auto response_json = nlohmann::json();
     // Returns a bad request response
-    auto const bad_request = [&req, &response_json](beast::string_view why) {
+    auto const bad_request = [&req, &response_json](boost::beast::string_view why) {
       response_json["error"] = std::string(why);
 
-      http::response<http::string_body> res{http::status::bad_request, req.version()};
-      res.set(http::field::server, BOOST_BEAST_VERSION_STRING);
-      res.set(http::field::content_type, "application/json");
+      // NOLINTNEXTLINE(cppcoreguidelines-init-variables)
+      boost::beast::http::response<boost::beast::http::string_body> res{boost::beast::http::status::bad_request,
+                                                                        req.version()};
+      res.set(boost::beast::http::field::server, BOOST_BEAST_VERSION_STRING);
+      res.set(boost::beast::http::field::content_type, "application/json");
       res.keep_alive(req.keep_alive());
       res.body() = response_json.dump();
       res.prepare_payload();
       return res;
     };
 
     // Make sure we can handle the method
-    if (req.method() != http::verb::get) {
+    if (req.method() != boost::beast::http::verb::get) {
       return send(bad_request("Unknown HTTP-method"));
     }
 
     // Request path must be absolute and not contain "..".
-    if (req.target().empty() || req.target()[0] != '/' || req.target().find("..") != beast::string_view::npos) {
+    if (req.target().empty() || req.target()[0] != '/' || req.target().find("..") != boost::beast::string_view::npos) {
       return send(bad_request("Illegal request-target"));
     }
 
-    http::string_body::value_type body;
+    // NOLINTNEXTLINE(cppcoreguidelines-init-variables)
+    boost::beast::http::string_body::value_type body{};
 
-    auto service_response = service_.GetMetrics().AsJson();
+    auto service_response = service_.AsJson(service_.GetMetrics());
```

Comment:
If we don't have any other format than JSON, wouldn't it be better to have only one method GetMetricsJSON? It is a bit weird to call two methods from same object. If we will need to support other formats, then it makes sense to separate functionalities but not atm

Reply:
Or at least AsJson method should be a static or some utility method since it doesn't relate in any way with object